src/Utility/FileLog.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Tools\Utility;
+
+use Cake\Log\Log as CoreLog;
+use Exception;
+
+/**
+ * Wrapper class to log data into custom file(s).
+ */
+class FileLog {
+
+	/**
+	 * Debug configuration.
+	 *
+	 * @var mixed|null
+	 */
+	protected static $_debugConfig;
+
+	/**
+	 * Initialize configurations.
+	 *
+	 * @param string $filename Filename to log.
+	 * @return void
+	 */
+	protected static function _init($filename) {
+		if ($filename === null) {
+			$filename = 'custom_log';
+		}
+
+		CoreLog::setConfig('custom', [
+			'className' => 'File',
+			'path' => LOGS,
+			'levels' => [],
+			'scopes' => ['custom'],
+			'file' => $filename,
+		]);
+
+		static::$_debugConfig = CoreLog::getConfig('debug');
+
+		CoreLog::drop('debug');
+	}
+
+	/**
+	 * Log data into custom file
+	 *
+	 * @param array|string $data Data to store
+	 * @param string|null $filename Filename of log file
+	 * @param bool $traceKey Add trace string key into log data
+	 * @return bool Success
+	 */
+	public static function write($data, $filename = null, $traceKey = false) {
```

Comment:
Do we want to use ?string typehint and bool one here?

Reply:
We are not returning string so only have to typehint `bool`